src/ecs/commands/updateEnableExecuteCommandFlag.ts

```diff
@@ -29,50 +27,49 @@ export async function updateEnableExecuteCommandFlag(
         'AWS.command.ecs.runCommandInContainer.warning.disableExecuteFlag',
         'Disabling command execution will change the state of resources in your AWS account, including but not limited to stopping and restarting the service.\n Continue?'
     )
+    const alreadyEnabled = localize('AWS.command.ecs.enableEcsExec.alreadyEnabled', 'ECS Exec is already enabled for this service')
+    const alreadyDisabled = localize('AWS.command.ecs.enableEcsExec.alreadyDisabled', 'ECS Exec is already disabled for this service')
     const yes = localize('AWS.generic.response.yes', 'Yes')
     const yesDontAskAgain = localize('AWS.message.prompt.yesDontAskAgain', "Yes, and don't ask again")
     const no = localize('AWS.generic.response.no', 'No')
+    
+    const prompt = enable ? 'ecsRunCommandEnable' : 'ecsRunCommandDisable'
+    const hasExecEnabled = node.service.enableExecuteCommand
 
-    if (enable) {
-        if (hasExecEnabled) {
-            window.showInformationMessage(
-                localize('AWS.command.ecs.enableEcsExec.alreadyEnabled', 'ECS Exec is already enabled for this service')
-            )
-            return
-        }
-        if (await settings.isPromptEnabled('ecsRunCommandEnable')) {
-            const choice = await window.showWarningMessage(enableWarning, yes, yesDontAskAgain, no)
+    const warningMessage = enable ? enableWarning : disableWarning
+    const redundentActionMessage = enable ? alreadyEnabled : alreadyDisabled
```

Comment:
Nit: alreadyEnabled/Disabled aren't necessary as separate consts since they're only used here. Feel free to move localize calls directly to here. You might even just want to have this statement directly in the showInformationWindow call since this is also only used here.

Reply:
I understand this reduces code slightly, but would that make it less readable?

Reply:
Condensed this a bit